src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -10,32 +12,53 @@
 
 (tk/defservice certificate-authority-service
   CaService
-  [[:PuppetServerConfigService get-config get-in-config]
-   [:WebroutingService add-ring-handler get-route]
-   [:AuthorizationService wrap-with-authorization-check]]
+  {:required [[:PuppetServerConfigService get-config get-in-config]
+              [:WebroutingService add-ring-handler get-route]
+              [:AuthorizationService wrap-with-authorization-check]]
+   :optional [FilesystemWatchService]}
```

Comment:
Originally we didn't make this optional, but then made it optional because we didn't expect it to work on all supported versions of Java. We no longer support versions of Java that we were concerned about so I think it makes sense to go back to how we were originally going to do it have this be mandatory.

iow, why would our users want to turn automatic CRL reloading off? and why would we let them?

Reply:
I think that's fair. Talking with @haus he also suggested we shouldn't make it optional in 5.x. My thinking behind leaving it as optional (and shipping it on by defaults) is that if for some reason this doesn't perform well or falls over under load, we can just tell people how to turn it off without needing to immediately ship them a new release

Reply:
(not that we do that with any other features, so i'm probably just being too nervous there)

Reply:
I wrote a big long thing about this, but it boiled down to: I think this is the kind of thing that is a good candidate for toggling, but we've already committed to shipping this on by default commercially for our customers that are most likely to be affected by it, so if we are concerned about its performance then I think we should be doing more testing there rather than making it toggle-able here.

Reply:
👍  that makes sense to me